### ISSUE TITLE
Support for custom/local models

### ISSUE THREAD

**Issue by ansarizafar** [run-llama/LlamaIndexTS#50]
### Details

Is it possible to use custom LLM like https://github.com/EvanZhouDev/bard-ai with Llamaindex?

**Comment by kevinlu1248** [comment]
sweep: Retry

**Comment by yisding** [comment]
@ansarizafar Yes this should be doable and is on our roadmap. Currently probably the easiest thing to do would be to extend the LLM interface to another class to support the local LLM.

```
export interface LLM {
  /**
   * Get a chat response from the LLM
   * @param messages
   */
  chat(messages: ChatMessage[], parentEvent?: Event): Promise<ChatResponse>;

  /**
   * Get a prompt completion from the LLM
   * @param prompt the prompt to complete
   */
  complete(prompt: string, parentEvent?: Event): Promise<CompletionResponse>;
}
```

Would you be interested in giving it a try?

**Comment by kevinlu1248** [comment]
Sweep: try again with the additional context from @yisding

**Comment by yisding** [comment]
Will be in the next version.

**Comment by richardtape** [comment]
Hi @yisding

First, thank you so much for contributing this. It's remarkable work.

I noticed in the closed PR #56 you mention that you want to support local models in a different way and that in this issue you were hoping it would be part of the next version. Is this something that is still on the radar? Or is it implemented (and if so, can you point me where I could find it?)

Again, many thanks!

**Comment by EmanuelCampos** [comment]
> Hi @yisding
> 
> First, thank you so much for contributing this. It's remarkable work.
> 
> I noticed in the closed PR #56 you mention that you want to support local models in a different way and that in this issue you were hoping it would be part of the next version. Is this something that is still on the radar? Or is it implemented (and if so, can you point me where I could find it?)
> 
> Again, many thanks!

Currently we have ollama that support local embeddings

**Comment by EmanuelCampos** [comment]
Closing due inactivity and for cleaning

Feel free to open again
